Neos.RedirectHandler.DatabaseStorage/Classes/Domain/Model/Redirection.php

```diff
@@ -0,0 +1,239 @@
+<?php
+namespace Neos\RedirectHandler\DatabaseStorage\Domain\Model;
+
+/*
+ * This file is part of the Neos.RedirectHandler package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Doctrine\ORM\Mapping as ORM;
+use TYPO3\Flow\Annotations as Flow;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+/**
+ * A Redirection model that represents a HTTP redirect
+ *
+ * @see RedirectionService
+ *
+ * @Flow\Entity
+ * @ORM\Table(
+ * 	indexes={
+ * 		@ORM\Index(name="targeturipathhash",columns={"targeturipathhash","hostpattern"})
+ * 	}
+ * )
+ */
+class Redirection
+{
+    /**
+     * @var \DateTime
+     */
+    protected $creationDateTime;
+
+    /**
+     * @var \DateTime
+     * @Gedmo\Timestampable(on="update", field={"sourceUriPath", "targetUriPath", "statusCode", "hostPattern"})
```

Comment:
IF we do this, then please move this package to the neos side of things, because there we require gedmo anyway. I don't really like the gedmo package and wouldn't want to introduce a dependency on it for Flow.